exercises/practice/mazy-mice/src/test/java/MazeGeneratorTest.java

```diff
@@ -0,0 +1,193 @@
+                .as("The maze is not null.")
+                .isNotNull();
+
+        Arrays.stream(maze)
+                .map(String::new)
+                .forEach(System.out::println);
+    }
+
+    @Ignore("Remove to run test")
+    @Test
+    void the_dimensions_are_correct() {
+        var maze = sut.generatePerfectMaze(RECTANGLE);
+
+        assertThat(maze)
+                .as("The dimensions of the maze are correct.")
+                .hasDimensions(RECTANGLE.height(), RECTANGLE.width());
+    }
+
+    @Ignore("Remove to run test")
+    @Test
+    void twoMazesShouldNotBeEqual() {
+        var maze1 = sut.generatePerfectMaze(RECTANGLE);
+        var maze2 = sut.generatePerfectMaze(RECTANGLE);
+
+        assertThat(maze1)
+                .as("Two mazes should not be equal")
+                .isNotEqualTo(maze2);
+    }
+
+    @Ignore("Remove to run test")
+    @Test
+    void twoMazesWithSameSeedShouldBeEqual() {
+        var maze1 = sut.generatePerfectMaze(RECTANGLE, 42);
+        var maze2 = sut.generatePerfectMaze(RECTANGLE, 42);
+
+        assertThat(maze1)
+                .as("Two mazes with the same seed should be equal")
+                .isEqualTo(maze2);
+    }
+
+    @Ignore("Remove to run test")
+    @Test
+    void twoMazesWithDifferentSeedsShouldNotBeEqual() {
+        var maze1 = sut.generatePerfectMaze(RECTANGLE, 42);
+        var maze2 = sut.generatePerfectMaze(RECTANGLE, 43);
+
+        assertThat(maze1)
+                .as("Two mazes with different seeds should not be equal")
+                .isNotEqualTo(maze2);
+    }
+
+    @Ignore("Remove to run test")
+    @Test
+    void theMazeContainsOnlyValidCharacters() {
+        var maze = sut.generatePerfectMaze(RECTANGLE);
+
+        for (var row : maze) {
+            for (var cell : row) {
+                assertThat(cell)
+                        .as("The maze contains only valid characters")
+                        .isIn(ALLOWED_SYMBOLS);
+            }
+        }
+    }
+
+    @Ignore("Remove to run test")
+    @Test
+    void theMazeHasOnlyOneEntranceOnTheLeftSide() {
+        var maze = sut.generatePerfectMaze(RECTANGLE);
+        int entranceCount = countEntrances(maze);
+
+        assertThat(entranceCount)
+                .as("The maze has only one entrance on the left side")
+                .isOne();
+    }
+
+    @Ignore("Remove to run test")
+    @Test
+    void theMazeHasASingleExitOnTheRightSideOfTheMaze() {
+        var maze = sut.generatePerfectMaze(RECTANGLE);
+        int exitCount = countExits(maze);
+
+        assertThat(exitCount)
+                .as("The maze has a single exit on the right side of the maze")
+                .isOne();
+    }
+
+    @Ignore("Remove to run test")
+    @Test
+    void theMazeIsPerfect() {
+        var maze = sut.generatePerfectMaze(RECTANGLE);
+
+        assertThat(maze)
+                .as("The dimensions of the maze are correct.")
+                .hasDimensions(RECTANGLE.height(), RECTANGLE.width());
```

Comment:
This assertion is already covered by another test case. I'd keep test cases as focused as possible, so since we are testing 'perfectness' of a maze, there is no need to assert on the dimensions as there are no requirements on dimensions for a perfect maze.

Reply:
done